server/src/main/java/org/apache/gravitino/server/web/filter/authorization/AssociateTagAuthorizationExecutor.java

```diff
@@ -0,0 +1,118 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.gravitino.server.web.filter.authorization;
+
+import static org.apache.gravitino.server.web.filter.ParameterUtil.buildNameIdentifierForBatchAuthorization;
+import static org.apache.gravitino.server.web.filter.ParameterUtil.extractFromParameters;
+
+import java.lang.reflect.Parameter;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.gravitino.Entity;
+import org.apache.gravitino.NameIdentifier;
+import org.apache.gravitino.authorization.AuthorizationRequestContext;
+import org.apache.gravitino.dto.requests.TagsAssociateRequest;
+import org.apache.gravitino.server.authorization.expression.AuthorizationExpressionEvaluator;
+import org.apache.gravitino.server.web.rest.MetadataObjectTagOperations;
+
+/**
+ * Metadata object authorization for {@link
+ * MetadataObjectTagOperations#associateTagsForObject(String, String, String, TagsAssociateRequest)}
+ */
+public class AssociateTagAuthorizationExecutor implements AuthorizationExecutor {
+
+  private final Parameter[] parameters;
+  private final Object[] args;
+  private final Map<Entity.EntityType, NameIdentifier> metadataContext;
+  private final AuthorizationExpressionEvaluator authorizationExpressionEvaluator;
+  private final Map<String, Object> pathParams;
+  private final String entityType;
+
+  public AssociateTagAuthorizationExecutor(
+      Parameter[] parameters,
+      Object[] args,
+      Map<Entity.EntityType, NameIdentifier> metadataContext,
+      AuthorizationExpressionEvaluator authorizationExpressionEvaluator,
+      Map<String, Object> pathParams,
+      String entityType) {
+    this.parameters = parameters;
+    this.args = args;
+    this.metadataContext = metadataContext;
+    this.authorizationExpressionEvaluator = authorizationExpressionEvaluator;
+    this.pathParams = pathParams;
+    this.entityType = entityType;
+  }
+
+  @Override
+  public boolean execute() throws Exception {
+    Object request = extractFromParameters(parameters, args);
+    if (request == null) {
+      return false;
+    }
+
+    AuthorizationRequestContext context = new AuthorizationRequestContext();
+    Entity.EntityType targetType =
+        Entity.EntityType.TAG; // Tags are the only supported batch target here
+
+    // Authorize both 'tagsToAdd' and 'tagsToRemove' fields.
+    if (request instanceof TagsAssociateRequest tagsAssociateRequest) {
+      return authorizeTagField(tagsAssociateRequest.getTagsToAdd(), context, targetType)
+          && authorizeTagField(tagsAssociateRequest.getTagsToRemove(), context, targetType);
+    }
+    throw new UnsupportedOperationException("Unsupported request type: " + request.getClass());
```

Comment:
Could you use `Precondition.checkArguments()`. This should throw illegalAgurmentException here.